tests/test_plugins/test_design.py

```diff
@@ -1,24 +1,61 @@
 """Test the parameter sweep plugin."""
 
+import sys
+
 import matplotlib.pyplot as plt
 import numpy as np
 import pytest
 import scipy.stats.qmc as qmc
 import tidy3d as td
 import tidy3d.web as web
 from tidy3d.plugins import design as tdd
-from tidy3d.plugins.design.method import MethodIndependent
 
 from ..utils import assert_log_level, run_emulated
 
 SWEEP_METHODS = dict(
     grid=tdd.MethodGrid(),
-    monte_carlo=tdd.MethodMonteCarlo(num_points=5),
-    custom=tdd.MethodRandomCustom(num_points=5, sampler=qmc.Halton(d=3)),
-    random=tdd.MethodRandom(num_points=5),  # TODO: remove this if not used
+    monte_carlo=tdd.MethodMonteCarlo(num_points=3, rng_seed=1),
+    custom=tdd.MethodRandomCustom(
+        num_points=5, sampler=qmc.Halton, sampler_kwargs={"d": 3, "seed": 1}
+    ),
+    random=tdd.MethodRandom(num_points=3, rng_seed=1),  # TODO: remove this if not used
+    bay_opt=tdd.MethodBayOpt(initial_iter=2, n_iter=2, rng_seed=1),
+    gen_alg=tdd.MethodGenAlg(solutions_per_pop=4, n_generations=2, n_parents_mating=2, rng_seed=1),
+    part_swarm=tdd.MethodParticleSwarm(n_particles=3, n_iter=2, rng_seed=1),
 )
 
 
+def emulated_batch_run(simulations, path_dir: str = None, **kwargs):
+    data_dict = {task_name: run_emulated(sim) for task_name, sim in simulations.simulations.items()}
+    task_ids = dict(zip(simulations.simulations.keys(), data_dict.keys()))
+    task_paths = dict(zip(simulations.simulations.keys(), simulations.simulations.keys()))
+
+    class BatchDataEmulated(web.BatchData):
+        """Emulated BatchData object that just returns stored emulated data."""
+
+        data_dict: dict
+        # task_ids: dict
+        # task_paths: dict
+
+        def items(self):
+            yield from self.data_dict.items()
+
+        def __getitem__(self, task_name):
+            return self.data_dict[task_name]
+
+    return BatchDataEmulated(data_dict=data_dict, task_ids=task_ids, task_paths=task_paths)
+
+
+def emulated_estimate_cost(self, verbose=True):
```

Comment:
I've refactored the testing and made a mock job.delete so the tests don't talk to the live servers. Tests read better now, this code is deterministic using pytest.parameterize with monkeypatch, and the 36 tests run in < 30 sec instead of > 5 mins.